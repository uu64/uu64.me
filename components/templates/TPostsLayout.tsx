import React from "react";
import { MDXProvider } from "@mdx-js/react";
import AList from "@/components/atoms/AList";
import ALink from "@/components/atoms/ALink";
import AImage from "@/components/atoms/AImage";
import AParagraph from "@/components/atoms/AParagraph";
import ABlockquote from "@/components/atoms/ABlockquote";
import APostHeading from "@/components/atoms/APostHeading";
import AInlineCode from "@/components/atoms/AInlineCode";
import MHead from "@/components/molecules/MHead";
import MCodeBlock from "@/components/molecules/MCodeBlock";
import OPostHeader from "@/components/organisms/OPostHeader";
import siteconfig from "@/siteconfig.json";

interface Props {
  frontMatter: FrontMatter;
  path: string;
  children: React.ReactNode;
}

const TPostsLayout: React.FC<Props> = (props: Props) => {
  const { children, frontMatter, path } = props;
  const state = {
    h1: APostHeading.H1,
    h2: APostHeading.H2,
    h3: APostHeading.H3,
    h4: APostHeading.H4,
    h5: APostHeading.H5,
    p: AParagraph,
    blockquote: ABlockquote,
    ul: AList.Ul,
    li: AList.Li,
    a: ALink,
    img: AImage,
    inlineCode: AInlineCode,
    code: MCodeBlock,
  };

  return (
    <div className="blog-post-content px-8">
      <MHead
        title={`${frontMatter.title} | ${siteconfig.name}`}
        description={frontMatter.description}
        pageUrl={`${siteconfig.url}${path}`}
        coverImageUrl={`${siteconfig.url}${frontMatter.cover}`}
        pageType="article"
      />
      <MDXProvider components={state}>
        <OPostHeader frontMatter={frontMatter} />
        {children}
      </MDXProvider>
    </div>
  );
};

export default TPostsLayout;
