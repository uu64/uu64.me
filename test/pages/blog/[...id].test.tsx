import React from "react";
import renderer from "react-test-renderer";
import TPostsLayout from "@/components/templates/TPostsLayout";
import MDX, { frontMatter as fms } from "@/posts/test/sample.mdx";

describe("App", () => {
  it("renders correctly", async () => {
    const frontMatter: FrontMatter = Array.isArray(fms) ? fms[0] : fms;
    const component = renderer.create(
      <TPostsLayout frontMatter={frontMatter} path="/blog/test/sample">
        <MDX />
      </TPostsLayout>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
