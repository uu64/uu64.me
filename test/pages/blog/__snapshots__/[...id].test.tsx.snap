// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="blog-post-content px-8"
>
  <div
    className="bg-custom-night1 p-8 rounded-xl"
  >
    <div
      className="py-8"
    >
      <div
        className="pb-2 text-custom-night3"
      >
        1970-01-01
      </div>
      <h1
        className="text-3xl text-custom-frost1 pb-4 border-b-2 border-custom-aurora4"
      >
        Sample
      </h1>
    </div>
    <h1
      className="py-12 text-custom-frost1 font-bold"
    >
      Heading 1
    </h1>
    <h2
      className="pt-12 text-custom-frost1 pb-4 font-bold"
    >
      Heading 2
    </h2>
    <h3
      className="pt-12 text-custom-frost1 pb-4 font-bold"
    >
      Heading 3
    </h3>
    <h4
      className="py-8 text-custom-frost1 font-bold"
    >
      Heading 4
    </h4>
    <h5
      className="py-8 text-custom-frost1 font-bold"
    >
      Heading 5
    </h5>
    <p
      className="leading-relaxed"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <blockquote
      className="ABlockquote__BlockQuote-sc-1chhqvt-0 iCfnfn border-l-8 border-gray-500 bg-gray-300 text-gray-700 px-4 mb-4"
    >
      <p
        className="leading-relaxed"
      >
        This is a blockquote element.
      </p>
    </blockquote>
    <ul
      className="ml-6"
    >
      <li
        className="list-disc"
      >
        item 1
      </li>
      <li
        className="list-disc"
      >
        item 2
      </li>
      <li
        className="list-disc"
      >
        item 3
        <ul
          className="ml-6"
        >
          <li
            className="list-disc"
          >
            sub item 1
          </li>
          <li
            className="list-disc"
          >
            sub item 2
          </li>
          <li
            className="list-disc"
          >
            sub item 3
          </li>
        </ul>
      </li>
    </ul>
    <p
      className="leading-relaxed"
    >
      <a
        className="text-custom-frost1 hover:underline"
        href="https://google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://google.com
      </a>
    </p>
    <p
      className="leading-relaxed"
    >
      <img
        alt="icon"
        className="py-4"
        src="/icon.jpg"
      />
      <span
        className="block text-center text-gray-600"
      >
        This is a sample image.
      </span>
    </p>
    <p
      className="leading-relaxed"
    >
      This is 
      <span
        className="bg-custom-night3 text-custom-snow0 font-mono text-sm rounded"
        style={
          Object {
            "padding": "0.2em 0.4em",
          }
        }
      >
        inline code
      </span>
      .
    </p>
    <pre>
      <pre
        className="prism-code language-bash font-mono overflow-x-auto -mx-8 my-4 p-4 text-sm"
        style={
          Object {
            "backgroundColor": "#292d3e",
            "color": "#bfc7d5",
          }
        }
      >
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#bfc7d5",
            }
          }
        >
          <span
            className="token shebang important"
            style={Object {}}
          >
            #!/bin/bash
          </span>
          <span
            className="token plain"
          >
            
          </span>
        </div>
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#bfc7d5",
            }
          }
        >
          <span
            className="token plain"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            

          </span>
        </div>
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#bfc7d5",
            }
          }
        >
          <span
            className="token plain"
          >
            
          </span>
          <span
            className="token builtin class-name"
            style={
              Object {
                "color": "rgb(255, 203, 107)",
              }
            }
          >
            echo
          </span>
          <span
            className="token plain"
          >
             Hello, World
          </span>
          <span
            className="token operator"
            style={
              Object {
                "color": "rgb(137, 221, 255)",
              }
            }
          >
            !
          </span>
          <span
            className="token plain"
          >
            
          </span>
        </div>
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#bfc7d5",
            }
          }
        >
          <span
            className="token plain"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            

          </span>
        </div>
      </pre>
    </pre>
  </div>
</div>
`;
